### PR TITLE
Oops. Fixes BaseController.js issue that broke the server

### PR DESCRIPTION
**Pull request by mglittleton**

### CODE REVIEW

**Review by BryceEvans — DISMISSED** [review]

**Review by BryceEvans — COMMENTED** [review]
I didn't see this, but it looks like there is a merge conflict in the Base Controller. You better fix that since you seem to have a better grasp of what is going on...

File: server/src/utils/BaseController.js [hunk 1]
```diff
@@ -38,13 +38,20 @@ export class BaseController {
...
+    let filter,
+      sort,
+      skip,
+      limit;
 
     if (req.query.filter) {
       filter = JSON.parse(req.query.filter);
     }
 
+    if (query) {
```
Comment (LBWright) [inline server/src/utils/BaseController.js:51]: an empty object is truthy so this statement will always hit

**Review by LBWright — CHANGES_REQUESTED** [review]

File: server/src/utils/BaseController.js (hunk 1 above)
Comment (LBWright) [reply server/src/utils/BaseController.js:51]: Maybe you could default it to null, or check to see if the object isn't empty?
Lodash has a good `isEmpty` function to use on objects.

Comment (mglittleton) [reply server/src/utils/BaseController.js:51]: Good point.

File: server/src/utils/BaseController.js [hunk 2]
```diff
@@ -44,9 +44,7 @@ export class BaseController {
       skip,
       limit;
     
-    if (query) {
-      filter = {...filter, ...query}
-    }
+    filter = {...filter, ...query}
 
     if (req.query.filter) {
```
Comment (LBWright) [inline server/src/utils/BaseController.js:47]: query will still be an empty object here, which will coerce `req.query.filter` into this `if` block

**Review by LBWright — CHANGES_REQUESTED** [review]

**Review by LBWright — APPROVED** [review]

**Pull request merged**